vxturn/turn.py

```diff
@@ -83,13 +84,29 @@ def get_send_params(self, message):
 
         return json.dumps(params).encode('ascii')
 
+    def verify_signature(self, content, signature, secret):
+        h = HMAC.new(secret)
+        h.update(content)
+        return h.hexdigest() == signature
 
     @inlineCallbacks
     def handle_raw_inbound_message(self, message_id, request):
         try:
-            # TODO: validate HMAC secret
-
-            payload = json.loads(request.content.read())
+            content = request.content.read()
+            headers = request.requestHeaders
+            try:
+                signature = headers.getRawHeaders('http_x_turn_hook_signature')[0]
```

Comment:
I think these `HTTP_...` things are only in django land, and here we need to use the direct `X-Turn-Hook-Signature`